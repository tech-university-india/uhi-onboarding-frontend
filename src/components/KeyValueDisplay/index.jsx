import React from 'react'
import Image from 'next/image'
import PropTypes from 'prop-types'

import editSvg from '../../../public/edit.svg'

export default function KeyValueDisplay ({ keyText, valueText, onEditClick }) {
  return <div className="p-4 w-full flex flex-row border-opacity-5 justify-between items-center border border-gray-100 bg-white drop-shadow-keyValueDisplay">
    <span className="drop-shadow-none">{keyText}</span>
    <div className="flex flex-row justify-end items-center">
      <span className="drop-shadow-none">{valueText}</span>
      {onEditClick &&
        <button className="pl-2 bg-transparent" onClick={onEditClick}>
          <Image src={editSvg} alt="Edit" />
        </button>
      }
    </div>
  </div>
}

KeyValueDisplay.propTypes = {
  keyText: PropTypes.string.isRequired,
  valueText: PropTypes.string.isRequired,
  onEditClick: PropTypes.func
}
