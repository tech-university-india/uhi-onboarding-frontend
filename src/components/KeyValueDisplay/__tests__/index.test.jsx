import React from 'react'
import KeyValueDisplay from '..'

import { render } from '@testing-library/react'

describe('KeyValueDisplay', () => {
  it('should render the component without onEditClick callback props', () => {
    const dummyPropsWithoutOnEditClick = {
      keyText: 'Key',
      valueText: 'Value'
    }

    const keyValueComponent = render(<KeyValueDisplay {...dummyPropsWithoutOnEditClick} />).asFragment()
    expect(keyValueComponent).toMatchSnapshot()
  })

  it('should render the component with onEditClick callback props', () => {
    const dummyPropsWithOnEditClick = {
      keyText: 'Key',
      valueText: 'Value',
      onEditClick: () => {}
    }

    const keyValueComponent = render(<KeyValueDisplay {...dummyPropsWithOnEditClick} />).asFragment()
    expect(keyValueComponent).toMatchSnapshot()
  })
})
