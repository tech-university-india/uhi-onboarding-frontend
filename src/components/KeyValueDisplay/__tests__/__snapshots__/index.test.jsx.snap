// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValueDisplay should render the component with onEditClick callback props 1`] = `
<DocumentFragment>
  <div
    class="p-4 w-full flex flex-row border-opacity-5 justify-between items-center border border-gray-100 bg-white drop-shadow-keyValueDisplay"
  >
    <span
      class="drop-shadow-none"
    >
      Key
    </span>
    <div
      class="flex flex-row justify-end items-center"
    >
      <span
        class="drop-shadow-none"
      >
        Value
      </span>
      <button
        class="pl-2 bg-transparent"
      >
        <img
          alt="Edit"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`KeyValueDisplay should render the component without onEditClick callback props 1`] = `
<DocumentFragment>
  <div
    class="p-4 w-full flex flex-row border-opacity-5 justify-between items-center border border-gray-100 bg-white drop-shadow-keyValueDisplay"
  >
    <span
      class="drop-shadow-none"
    >
      Key
    </span>
    <div
      class="flex flex-row justify-end items-center"
    >
      <span
        class="drop-shadow-none"
      >
        Value
      </span>
    </div>
  </div>
</DocumentFragment>
`;
